### ISSUE TITLE
0.57.0 Discussion

### ISSUE THREAD

**Issue by grabbou** [react-native-community/react-native-releases#34]
Conversation on this thread are limited to 0.57 releases major issues and backport (cherry-pick) requests from commits that are **already** on master.

An example of a good such request is a bug fix for a serious issue that has been merged into master but did not make the 0.57 RC cut.

In other words, if you cannot point to a particular commit on master, then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

**Comment by kelset** [comment]
Before we reach 0.57.0 we need to be sure this is fixed/closed: https://github.com/facebook/react-native/issues/20327

**Comment by Kevat** [comment]
Will this issue be resolved by 0.57:
https://github.com/facebook/react-native/issues/19953

When is 0.57 due for release? It has been frustrating not being able to use react native on Windows.

**Comment by jamsch** [comment]
@kelset I believe you need to set `module:metro-react-native-babel-preset` if the module isn't prefixed with "babel-preset".

Also preferably [this commit](https://github.com/facebook/react-native/commit/892212bad2daadd373f4be241e4cd9889b0a1005) should be cherry picked in to the next release, it's one of the few issues that we're having to maintain a fork for.

**Comment by kelset** [comment]
Hey thanks for the feedback - we still haven't released the first rc because we had some issues testing RNTester, so we'll basically sync the 0.57 branch with master asap so you'll also get that fix (I want it too 😅).

**Comment by rafaellincoln** [comment]
Is there any chance of this [commit](https://github.com/facebook/react-native/commit/da6a5e0439c168147271ef66ad5ebbeebd6fce3b)  entering version 0.57?

**Comment by kelset** [comment]
@rafaellincoln we are cutting the first RC of 0.57 today (unless during our testing something weird happens) and we'll basically synch the branch with master first, so you'd be able to see that commit in it 💪

**Comment by fungilation** [comment]
Update on 0.57 please? I'm not seeing any RC on https://github.com/facebook/react-native/releases

**Comment by kelset** [comment]
While testing the branch we got stuck on this https://github.com/facebook/react-native/issues/20567#issue-348388441

so until we resolve it we won't release.

**Comment by kelset** [comment]
Quick update, we managed - with the help of a few amazing devs of this community - to finally get to a stage where we are feeling ok with releasing the first RC.

Currently CI is blocked by an invalid npm token, but it's being taken care of so expect 0.57-rc0 to be on npm in the next 12/24 hours.

**Comment by grabbou** [comment]
While we are waiting for the `npm` token to be released, there's work already happening on getting the docs in place (thanks to @turnrye).

We are looking for a period of ~2 weeks before we promote RC to stable. That should give us enough time to make sure it works without any major regressions and get us back into the monthly release cycle.

Note: The release cycle is being currently discussed [here](https://github.com/react-native-community/discussions-and-proposals/issues/17)

**Comment by abinax** [comment]
react native 0.57-RC has gone from https://facebook.github.io/react-native/versions
Why?

**Comment by grabbou** [comment]
React Native 0.57.0-rc.0 is out https://www.npmjs.com/package/react-native/v/0.57.0-rc.0. The docs are going to be published in a while I believe.

**Comment by fungilation** [comment]
Can't go further than `react-native-git-upgrade 0.57.0-rc.0` since that's [breaking](https://github.com/facebook/react-native/issues/20710) for me.

**Comment by kelset** [comment]
A few issues and PRs we should try to have merged for 0.57.0:
- https://github.com/facebook/react-native/issues/20703 fixed via https://github.com/facebook/react-native/pull/20706
- https://github.com/facebook/react-native/pull/20705
- https://github.com/facebook/react-native/pull/20662
- we should make sure to have metro version that uses this https://github.com/facebook/metro/pull/223

**Comment by danielgindi** [comment]
Is 0.57.0-rc.0 published to nexus?
Because there are issues with many native libraries, with dependencies like `compile 'com.facebook.react:react-native:+'` - where Android Studio tries to download 0.20.1 and compile against that.

**Comment by rozele** [comment]
Once it's merged, can we cherry-pick https://github.com/facebook/react-native/pull/20705 onto 0.57-stable and cut another RC for `react-native-windows` and `react-native-dom`

**Comment by janhesters** [comment]
Will [this issue #19859](https://github.com/facebook/react-native/issues/19859) (testing) be fixed in version 0.57? It feels weird for a lot of new people that testing is broken out of the box.

**Comment by grabbou** [comment]
Yes, @janhesters, this should be fixed. I already commented on that issue. Thanks for rising that - this is exactly what this issue is for.

@danielgindi, please raise your issue in RN repository itself. The RN comes always from `node_modules` and we never publish the Android version to any external dependency center (including Maven etc.). I believe there might be some historical versions published there, which gives you an error about the wrong version. Try installing node modules again.

@fungilation there's ongoing discussion about whether to rewrite/drop `react-native-git-upgrade` entirely due to its brittleness with new upgrades. Are you able to provide your feedback about it, preferably via email? (it's on my profile) We could start a proposal to create a better, new, version of it. If you feel like having enough free time, I could help you get it up and running.

**Comment by janhesters** [comment]
@grabbou Awesome news! Thank you very much.

**Comment by kelset** [comment]
Update on my previous comment:
* one of @rozele's PRs landed, so we need to cherry-pick this https://github.com/facebook/react-native/commit/c5297c75cb6da58a241c8f91b0d2fefbc5835a46 and wait for Metro 0.43.7 to be releases since this commit is not published yet https://github.com/facebook/metro/commit/387d487bbc44afebc1710c5b2869121ba3b64030
* until Metro 0.43.7 is released this can't be merged https://github.com/facebook/react-native/pull/20706 (and cherry pick it)
* we'll also need to cherry pick this https://github.com/facebook/react-native/pull/20751 (context: https://github.com/facebook/react-native/issues/20712#issuecomment-414075393 )

To answer @janhesters - if you've read my [comment at the bottom](https://github.com/facebook/react-native/issues/19859#issuecomment-407748189) in that issue, it explains how it's not a react-native issue per se. That said we may consider "removing" jest from the init template so that we can avoid this:
> It feels weird for a lot of new people that testing is broken out of the box.

**Comment by rozele** [comment]
@kelset - I commented on facebook/react-native#20706. That comment does not depend on the change to `metro`. Metro still accepts the option for `--reset-cache`, the change to `metro` only affects the internal handling of that flag and ensures it's passed down to `jest-haste-map`

**Comment by fungilation** [comment]
@grabbou I'm emailing you. But I'll say here that `react-native-git-upgrade` has been reliably upgrading every version for me since about 0.50 or earlier. And no doubt it's better than `react-native upgrade` with its 3 way merging of template files, instead of manual diffing. I'm all for a better version of `react-native-git-upgrade`.

**Comment by janhesters** [comment]
@kelset Yup I've read your comment and it helped me tremendously! It's a nicely working fix.

Why not just add that fix to the stable version of React Native? In my opinion removing jest from `react-native init` template would be a bad move, because having Jest shipped with React Native encourages great programming!
I always loved for example how Django also ships with testing files and has a explicit chapter about testing in it's docs.
When I first picked up React Native I was excited to see that other frameworks also animate the user to write tests. It's an amazing privilege for new programmers, when they get lead by more experienced programmers to use best practices and write long-term stable code 🔥🤓

**Comment by hramos** [comment]
A stack of commits landed on master a few days ago, which bring support for `WKWebView`-backed `WebView`s in React Native. With the deprecation of `UIWebView` in iOS 12, and 0.57 likely to reach stable right around the same time as iOS 12 is released to the public, I believe it's worth cherry-picking all of these onto `0.57-stable`:

- https://github.com/facebook/react-native/commit/1442c265da36601114ce184cd5bc322f45dc1b44
- https://github.com/facebook/react-native/commit/3703927e7e12ffc8922644ea251cd6f7a384570c
- https://github.com/facebook/react-native/commit/7a6dd9807cda45c2d60641864f2d6c8d401e8ae3
- https://github.com/facebook/react-native/commit/e5f95aba9b23376de498456282ad17113ef44cd9
- https://github.com/facebook/react-native/commit/1741fe95710556f30dc2442aaaae23e31dad4cc0
- https://github.com/facebook/react-native/commit/90e85a4adc749666f81034119f281ac54840e7df
- https://github.com/facebook/react-native/commit/0022354525eae0a368704da65c9d0f85f33ba5fb
- https://github.com/facebook/react-native/commit/03b57d9db6509fa3e715f23c8270caf6ca091acd
- https://github.com/facebook/react-native/commit/1584108805ca6c8eff7a77e15c8553028665b53f
- https://github.com/facebook/react-native/commit/a997c0ac16d8863333d057269a8b5e28994b84eb
- https://github.com/facebook/react-native/commit/4ca949b46ec8fd72b5305daa06fac3ef58a8fa5f
- https://github.com/facebook/react-native/commit/721763020a4a7b4b3cad1a9c074ec2e51a8d840b
- https://github.com/facebook/react-native/commit/1af17f164828b6d6fa0450af46baf945745363e7
- https://github.com/facebook/react-native/commit/215fa14efc2a817c7e038075163491c8d21526fd
- https://github.com/facebook/react-native/commit/bacfd9297657569006bab2b1f024ad1f289b1b27
- https://github.com/facebook/react-native/commit/95801f1eda2d723d9b87760d88fa9f1a1bb33ab1
- https://github.com/facebook/react-native/commit/b18fddadfeae5512690a0a059a4fa80c864f43a3
- https://github.com/facebook/react-native/commit/28b058c341690bd35e1d59885762ec29614a3d45
- https://github.com/facebook/react-native/commit/33b353c97c31190439a22febbd3d2a9ead49d3c9

We have a blog post scheduled which will talk about these changes. People can opt in to the `WKWebView` through a new `useWebView` prop, so this should be a relatively safe change.

**Comment by fungilation** [comment]
Great news with moving to/adding WKWebView!

This is likely offtopic, but I'm stuck using abandoned https://github.com/alinz/react-native-webview-bridge due to this: https://github.com/facebook/react-native/pull/9762#issuecomment-260583009. Would be great to allow setting `window.postMessage` to override any existing one in an arbitrary webpage, or avoid name collision by moving RN's injection to `window.rnPostMessage`. Had a long discussion before on [#9762](https://github.com/facebook/react-native/pull/9762#issuecomment-260583009) and don't remember where else on this and it didn't go anywhere.

Where should a discussion on this be had? Alongside or after the work on adding WKWebView, which as I understand it has new APIs which makes implementing this easier? I'm willing to investigate and implement if lack of interest is the issue with having a functional, non-colliding `window.postMessage`. I just need direction on how to approach getting this into RN, if it's wanted. As I need this in my app, and I want to move off unmaintained `react-native-webview-bridge`.

**Comment by kelset** [comment]
👋 thanks for the comments, I believe that it may be worth creating an rc1 as soon as a PR for this workaround is created&merged (https://github.com/facebook/react-native/issues/20710#issuecomment-414631827).

**Comment by grabbou** [comment]
I agree, time for next RC once that one lands. Generally speaking I would
be against cherry-picking all the WebView related work in scope of this
release, but given the timing and possible delays that can occur while
releasing next stable release of React Native, I think it's a safe move as
explained by Hector.

**Comment by SudoPlz** [comment]
Guys I hope this comment is not off topic, but it's really worrying me.
0.57.* is about to be broadly used and this serious issue is yet not handled.

https://github.com/facebook/react-native/issues/8615

As far as I know NetInfo is malfunctioning and I have no idea what makes it miss-behave.
If there's another place to leave that comment let me know and I'll deleted from this thread, but it seems like this thread is generic discussion about 0.57

Thank you.

**Comment by kelset** [comment]
@SudoPlz as stated in the first comment of this issue
> In other words, if you cannot point to a particular commit on master, then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

That said, I also suggest you check [this comment](https://github.com/facebook/react-native/issues/8615#issuecomment-389358993) if that issue is still present for you.

**Comment by ngraef** [comment]
Please cherry-pick https://github.com/facebook/react-native/commit/9a77ff593bf83a3326e6237bb877e7abc437da96 to fix the bundler regression from https://github.com/facebook/react-native/issues/20712.

Also, thank you for rc2. The rc1 build was all kinds of broken.

**Comment by hramos** [comment]
@assafb81 this has been mentioned several times - please limit requests to cherry-picks from commits on master. Any future comments of this type will be hidden without explanation.

---

0.57.0-rc.1 became briefly available today, but it was missing Android artifacts. It was un-published, and 0.57.0-rc.2 is now on npm with all cherry-picks requested up to @kelset's comment above.

I'll cherry-pick facebook/react-native@9a77ff5 in order to get it into the next RC.

**Comment by assafb81** [comment]
@hramos thanks for that...did not see @kelset comment earlier.
I will close the new opened issue

**Comment by kelset** [comment]
Thanks @hramos!
Aside from those 2 commits, I've cherry picked https://github.com/facebook/react-native/commit/79fe925f1daa053d5a5d92a228e5c7beff565ab4

Let's see if anything else pops out from devs using rc2, if not I feel like we can release rc3 soon-ish.

**Comment by MarkOSullivan94** [comment]
Would love it if this could be included in rc3 - https://github.com/facebook/react-native/commit/6e356895e79fb92640295a14483af1a430732247

It means we can start making PRs to upgrade all of the RN plugins to use the new gradle keywords instead.

**Comment by kelset** [comment]
I tested locally the branch with the "new" 3 commits mentioned above
* https://github.com/facebook/react-native/commit/79fe925f1daa053d5a5d92a228e5c7beff565ab4
* https://github.com/facebook/react-native/commit/e61176d650e2b5fe51dd6cd4c429ff47a1a9b1dc
* https://github.com/facebook/react-native/commit/9a77ff5

and on macOS now (thanks to @rafeca's commit!) everything seems to be working fine, the `run-XXX` not spawning properly the packager is fixed.

So I think we can proceed to do a `rc3` release.

Related to @MarkOSullivan94 's request, in order to cherry pick that commit it's actually necessary (to avoid conflicts) to cherry pick:
* https://github.com/facebook/react-native/commit/874cca1ac258ec224bade999722d7a34c307def0
* https://github.com/facebook/react-native/commit/f884a1be69def7a9d629fc21596a4ec61a76ad4e
* https://github.com/facebook/react-native/commit/7ff9352a885ce450cfa0d2ce65d0b55e9774bd67
* https://github.com/facebook/react-native/commit/044b399e6590d84065a9b186750f77bc9d851aac
* (and finally) https://github.com/facebook/react-native/commit/6e356895e79fb92640295a14483af1a430732247

That said, I also tested locally the branch with those extra 5 commits and everything is still ✅ on my side, so unless @hramos or anyone else has strong opinions against, I feel like we could add those 5 commits and then proceed with `rc3`.

**Comment by hramos** [comment]
I'm OK with cherry-picking these, if `test_android` remains green after adding them to `0.57-stable`. It's currently red on master (https://circleci.com/gh/facebook/react-native/50086), and I have yet to investigate which specific commit introduced the regression.

**Comment by nadinelyab** [comment]
Seeing as other webview commits were cherry picked, could we include [facebook/react-native@e6b305b](https://github.com/facebook/react-native/commit/e6b305b722a8f52fad8ac8bf9f95204fa2192944)?

It would mean having webviews running injected javascript work much better in android

**Comment by fungilation** [comment]
I want to add to @nadinelyab in advocating https://github.com/facebook/react-native/commit/e6b305b722a8f52fad8ac8bf9f95204fa2192944 to be cherry picked. I have hit multiple cases on Android of injected JS on Android not working which this PR describes as fixing. Breakage on Android webview that works on iOS in my app. Having these fixed with a single PR would be much appreciated.

**Comment by kelset** [comment]
Yeah I think we can cherry pick that.
Glancing over the master commits, I think these may be also worth considering:
* https://github.com/facebook/react-native/commit/e2f2e41f097a128b7a89f20bed916394666d4670
* https://github.com/facebook/react-native/commit/3b688ae6b294e94f11e8eff425adf698e5b3783c

And these two commits by @rafeca:
* https://github.com/facebook/react-native/commit/dd0900a03af15c7d41c1e8b548410584e0abd01b
* https://github.com/facebook/react-native/commit/e4621f4ce142c23067ef5ab83465c01ebce34ff4

**Comment by hramos** [comment]
### [0.57.0-rc.3](https://www.npmjs.com/package/react-native/v/0.57.0-rc.3) is now available.

I've cherry-picked everything requested since rc.2, with a few exceptions:

- https://github.com/facebook/react-native/commit/e6b305b722a8f52fad8ac8bf9f95204fa2192944 was requested by @nadinelyab and @fungilation, but that commit has been a part of the RC since rc.0. 
- https://github.com/facebook/react-native/commit/e4621f4ce142c23067ef5ab83465c01ebce34ff4 requires a new version of Metro to be deployed to npm, so it will have to wait for the next release

I think we're at a point where we can focus on making sure there's no major blockers in this release, so we can promote 0.57 to stable soon.

---

As always, let's keep the discussion focused on getting 0.57 out the door -- avoid commenting on this thread unless you're requesting a cherry pick of a certain commit that's already on `master`. The best way to prioritize getting something fixed is by sending a PR to the RN repo.

**Comment by fungilation** [comment]
On https://github.com/facebook/react-native/commit/e6b305b722a8f52fad8ac8bf9f95204fa2192944, I can confirm though not extensively tested that `/* comment */` blocks are no longer needed in my app for injected JS inside webview, `// comment` works now just like in iOS.

I'm on rc.0

**Comment by hramos** [comment]
The commit https://github.com/facebook/react-native/commit/1f88a7111d1a64910a4bfab322820406250081dc is getting reverted on master soon. This commit is in the 0.57 RC, therefore we should cherry-pick the corresponding revert commit once it lands on master.

I'll edit my comment once the revert commit is on master.

**Comment by janhesters** [comment]
Would be awesome to include Babel 7 out of the box for 0.57, since Babel 7 supports Typescript :)

**Comment by emin93** [comment]
@janhesters The current changelog for the 0.57 release says that there is now integrated TypeScript support in the Metro bundler through Babel 7: https://github.com/react-native-community/react-native-releases/blob/0.57/CHANGELOG.md

**Comment by Titozzz** [comment]
Alright so metro@0.45 is now on master thanks to @rafeca with https://github.com/facebook/react-native/commit/169812f9ce60317dd7320384007879be16278678 we can cherry pick it and create the rc4 @hramos :tada:

**Comment by vshab** [comment]
It would be nice to see https://github.com/facebook/react-native/commit/fd744dd56ca183933a67e8398e1d20da14a31aab in this release.
https://github.com/facebook/react-native/issues/20155 is quite annoying currently. 

Thanks.

**Comment by grabbou** [comment]
Cherry-picked, waiting for CI and will release `rc4`

**Comment by Salakar** [comment]
https://github.com/facebook/react-native/pull/20854 - has been synced with master and all CI stages passing now 👌

**Comment by Titozzz** [comment]
As the RC4 deploy failed due to 
<img width="1264" alt="capture d ecran 2018-08-31 a 18 03 24" src="https://user-images.githubusercontent.com/6181446/44923388-65162200-ad48-11e8-8d2c-b78185cded7c.png">

Maybe we can cherry pick https://github.com/facebook/react-native/pull/20854 and release the RC4 with, hopefully all the commits we will need for React Native 0.57? Also, is there anything we can do to fix this step on Circle CI, or is this facebook internal?

**Comment by grabbou** [comment]
I haven't actually tagged `rc.4` yet, so this step was expected to fail anyway. This failure seems very suspicious and should be fixed (CC: @hramos how you were doing that previously?)

Going to cut it now and see where it goes.

**Comment by Salakar** [comment]
@Titozzz running `ssh-keyscan github.com >> ~/.ssh/known_hosts` might work to resolve that issue - whether that's needed as part of the script I'm not sure.

Something like:

```js
exec('ssh-keyscan github.com >> ~/.ssh/known_hosts', {silent: true});`
```

before line 65 of the `publish-npm.js` script maybe.

**Comment by christianbach** [comment]
Hi! It would be cool to see facebook/react-native@1cc29c5 in the next RC cut.

**Comment by sunnylqm** [comment]
@wangshangming Here is not a place to discuss a specific issue. You should just ask questions and wait in the original thread (and I saw some workarounds there already).

**Comment by ChrisEdson** [comment]
RC4 seems to be tagged - but no release on NPM?

**Comment by grabbou** [comment]
We are having issues with CircleCI not building correctly. Facebook team
will investigate this case in the meantime, going to leave the tag for now.
Thread to be updated once we fix the issue.

**Comment by Salakar** [comment]
@grabbou https://github.com/facebook/react-native/pull/20854 addressed a couple CI issues - or are these other CI issues?

**Comment by Titozzz** [comment]
They are having issues with the new NDK, @Salakar if you want to follow what is happening on a branch I use this syntax `https://circleci.com/gh/facebook/react-native/tree/<branch_name>`, for example https://circleci.com/gh/facebook/react-native/tree/0.57-stable

**Comment by birkir** [comment]
@Titozzz It's been stale for two days actually.

**Comment by sercand** [comment]
It seems react-native docker build image is `circleci/android:api-27-node8-alpha` but there is also 
`circleci/android:api-27-ndk-r17b` at  [docker hub](https://hub.docker.com/r/circleci/android/tags/). `api-27-node8-alpha` [don't have NDK in it](https://github.com/CircleCI-Public/circleci-dockerfiles/blob/d76304a20f636e7c92943559b09494b08fa47ee0/android/images/api-27-node8-alpha/Dockerfile) but `api-27-ndk-r17b` [don't have node and yarn](https://github.com/CircleCI-Public/circleci-dockerfiles/blob/d76304a20f636e7c92943559b09494b08fa47ee0/android/images/api-27-ndk-r17b/Dockerfile). Therefore react-native can't use both of them.

**Comment by birkir** [comment]
They have a circleci task to install NDK on the node8 image

**Comment by sercand** [comment]
@birkir You are correct it has [Install Android NDK](https://circleci.com/gh/facebook/react-native/51310) setup on `test_android` step but not in the `publish` step.

**Comment by fungilation** [comment]
Any update from Facebook people on how much longer this is going to stall?

**Comment by hramos** [comment]
I was out sick the past week, so I missed all of these updates. I'm re-tagging 0.57.0-rc.4 and having it go through Circle again.

**Comment by birkir** [comment]
Thanks @hramos - It went through, rc-4 is out

**Comment by hramos** [comment]
https://github.com/facebook/react-native/commit/1cc29c5 has been cherry-picked, but didn't make it into rc.4.

**Comment by fungilation** [comment]
Any difference in upgrade procedure, coming from rc1? Upgrading with `react-native-git-upgrade`, do we still need this in package.json dependencies: `"metro-react-native-babel-preset": "0.43.5",` and

```json
  "resolutions": {
    "@babel/core": "7.0.0-beta.56",
    "babel-core": "7.0.0-bridge.0"
  }
```
?

**Comment by kelset** [comment]
Thanks for taking care of that @hramos!

@fungilation I think you should only need the bridge version, and you should check the package.json for making sure which version of Metro is the latest on branch.

When we will release 0.57.0 (which is, hopefully, soon, based on the feedback on `rc4`) we'll make sure to include a list of manual steps to upgrade your apps.

**Comment by wangshangming** [comment]
@kelset Will it possible to replace global babelHelper and @babel-external-helper to @babel/runtime for better compatibility to third-party library since RN0.56.0+ is using babel7, some helpers such as applyDecoratedDescriptor is not defined in global babelHelper

**Comment by kelset** [comment]
@wangshangming could you open a PR for that? I think that it would be easier to visualise and understand how it affects the codebase.

That said unless it fixes a precise bug that is a release-blocker I don't think it will make it for 0.57.0. But maybe for a future minor release.

**Comment by janhesters** [comment]
This might be a nooby question, but how can I find the git repository for `rc4`? 😊

**Comment by Aidurber** [comment]
@janhesters Just the tag in the react-native repo: https://github.com/facebook/react-native/tree/v0.57.0-rc.4

**Comment by newyankeecodeshop** [comment]
@wangshangming @kelset There is a PR open in metro to replace babel helpers with the babel runtime. Hopefully it will be included in the next major release of metro?
https://github.com/facebook/metro/pull/198

**Comment by wangshangming** [comment]
@newyankeecodeshop yeah that's a way to solve the issue but still a better way to remove the babelHelper/regeneratorRuntime since it can not always catch up with the latest babel-helper

**Comment by kelset** [comment]
thanks for the link! Anyway let's keep this conv about existing commits and/or issues potentially blocking the release - again, @wangshangming, a PR would be the best way to discuss it :)

**Comment by Salakar** [comment]
Anyone else getting duplicate module issues on rc4? Looks like `metro-visualizer` installed as part of `react-native` causing it:

```
(node:91698) UnhandledPromiseRejectionWarning: 

Error: jest-haste-map: @providesModule naming collision:
  Duplicate module name: react

  Paths: 

	<PROJ_ROOT>/node_modules/react/package.json 
		collides with 
	<PROJ_ROOT>/node_modules/rc-hammerjs/node_modules/react/package.json

This error is caused by a @providesModule declaration with the same name across two different files.
    at setModule (<PROJ_ROOT>/node_modules/jest-haste-map/build/index.js:462:17)
    at workerReply (<PROJ_ROOT>/node_modules/jest-haste-map/build/index.js:512:9)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

#### `yarn why rc-hammerjs` outputs:
![image](https://user-images.githubusercontent.com/5347038/45231765-0d863200-b2c5-11e8-8d0a-a884b146e339.png)


Should `react-native` be more explicit as to what `metro` packages it requires rather than blanket requiring all of `metro` which adds all the metro packages ([`package.json`](https://github.com/facebook/metro/blob/master/packages/metro/package.json))?

**Comment by Titozzz** [comment]
@salakar I've updated my app to 0.57.0-rc.4 from 0.56 with _nearly_ no issues. The only duplicate I had is due to the fact that I have multiple react-native In my workspace yet I manage to deal with the blacklistRE param in the metro.config.js.

I'd like to make sure that 0.57 release is as smooth as possible for everyone, can you DM me on twitter so that we can find out what's wrong with your setup without polluting this thread too much 😅 ?

**Comment by Salakar** [comment]
@Titozzz have DM'd you - thanks 👍

**Comment by yurykorzun** [comment]
Why babel 6 is installed along with babel 7?

yarn why babel-core

[1/4] Why do we have the module "babel-core"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "babel-core@6.26.3"
info Reasons this module exists
   - "react-native#metro" depends on it
   - Hoisted from "react-native#metro#babel-core"
   - Hoisted from "react-native#fbjs-scripts#babel-core"
   - Hoisted from "jest#jest-cli#jest-runtime#babel-core"
   - Hoisted from "jest#jest-cli#jest-config#babel-core"
   - Hoisted from "react-native#metro#babel-register#babel-core"
   - Hoisted from "react-native#metro#metro-visualizer#emotion#babel-plugin-emotion#babel-core"

**Comment by sijad** [comment]
any chance facebook/react-native#14244 be resolved by 0.57?
this cause problems on old android (4.4) devices, we can not use cloudflare CDN because of this.

**Comment by fungilation** [comment]
@yurykorzun that's interesting. I've upgraded (by react-native-git-upgrade) to rc4, and

`@babel/core` in my yarn.lock shows 7.0.0, but
`babel-core` shows multiple 6.x deps that resolves to 6.26.3.

I can run android emulator fine on Windows 10 with this though.

I searched through RN's github repo, and the only reference to "babel-core" (not babel/core) is in https://github.com/facebook/react-native/blob/master/react-native-git-upgrade/package.json, which specifies ^6.18.0.

@kelset, is this correct or should it be version bumped (for git-upgrade and manual upgrade)? To `"babel-core": "7.0.0-bridge.0"` or just `"babel-core": "7.0.0"`?

**Comment by fungilation** [comment]
The bigger question before even where it's from, is what is the intended/correct version(s) post-install for babel/core, babel-core.

**Comment by LinusU** [comment]
The correct version for React Native 0.57 should be 7.x.x. As of rc.4 we upgraded to the stable Babel 7.

**Comment by Titozzz** [comment]
I think what is really important is that your build is done with Babel 7 and that Babel 6 does not conflict with that. If then we still have a dependency such as a cli that require babel-core or babel-runtime to run, that's fine and it can get bumped later if needed.

**Comment by birkir** [comment]
https://github.com/ueno-llc/react-native-starter

Fully working with typescript and everything, literally only thing I had to do was bump the version and modify the metro config a bit.

Try removing the npm/yarn lockfile and install again with just babel 7 (no bridge)

**Comment by kelset** [comment]
@fungilation `"babel-core"` is the old implementation, the new one is `"@babel/core"`. The reason why we suggest to have `"babel-core": "7.0.0-bridge.0"` it's because it's a [special version](https://github.com/babel/babel-bridge) that should help you handle scenarios in which you may have old packages that still use babel 6.

That said, again, pls keep this issue solely on the topic of the release and eventual commits to cherry pick.

**Comment by matt-block** [comment]
Since we updated Gradle, should we also include [facebook/react-native@`4dfdec9`](https://github.com/facebook/react-native/commit/4dfdec9b28074b311ffd75084d9d2a3cbd41a800) so to release 0.57 with all Gradle parts updated?

Then again, it's not a priority since the old `compile` configuration will be removed only at the end of the year in a future update.

**Comment by IljaDaderko** [comment]
Hey guys, since `0.57.0-rc.4` added support for "out of the box typescript" I decided to give it a go and stumbled across following issue:

It seems that it compiles `.ts` and `.tsx` files fine, however I have [module resolution](https://www.typescriptlang.org/docs/handbook/module-resolution.html) set up in my repo and am getting following error:

> Loading dependency graph, done.
> error: bundling failed: Error: Unable to resolve module `@myApp/services/validation` from `/Users/me/Documents/Repositories/myApp/packages/myApp/src/store/AuthStore.ts`: Module `@myApp/services/validation` does not exist in the Haste module map

This used to work with typescript setup pre `0.57.x`. Am I wrong in assuming that rn will respect existing `tsconfig.json` files as default or is additional configuration in babel needed for this?

**Comment by yurykorzun** [comment]
I decided to try to narrow down the problem I was having with 0.57 rc4 and try to find where the exception Unexpected token '...' was coming from. After adding react-navigation to the empty project I get the same error right away. It looks like react-navigation cannot resolve the object rest spread operator plugin. It works with RN 0.54

This is probably off topic and I will post it to react-navigation issues.

**Comment by grabbou** [comment]
@matt-block yes, we should. Unfortunately, it raises one important concern that I have - I described it in the commit. Let's look at it and discuss it for a moment before we decide to cherry-pick it. I am afraid it might be deal breaking for existing users of React Native `link`. 

That said, I could potentially spend some time tomorrow getting this fixed, as we have to ship it (who wants stable release with warnings), yet we need this one fixed and fully supportive of both formats, at least for one release.

**Comment by ikesyo** [comment]
https://github.com/facebook/react-native/commit/0bf56e0746c1e60204996558b15ba425b12ed457 must be cherry-picked to make the new `WKWebView` backend usable (the feature is listed in the 0.57 changelog's `Highlights` section).

**Comment by rowinbot** [comment]
Can we get [facebook/react-native@2307ea6](https://github.com/facebook/react-native/commit/2307ea60d03edd234511bfe32474c453f30c1693) cherry-picked into next release? 

Actually, we're unable to mask an input or append a validation 'cuz TextInput is not taking the value being passed as a prop as the actual input value.

**Comment by jonathanglasmeyer** [comment]
Just tracked down an issue while trying to get a running project with RC4: the react-native local-cli fails with 
```
ReferenceError: regeneratorRuntime is not defined
    at getCliConfig (/Users/jwerner/projects/rn-test/Foo/node_modules/react-native/local-cli/core/index.js:128:44)
```
when babel is setup with `babel.config.js`:
```js
// babel.config.js
module.exports = {
  "presets": ["module:metro-react-native-babel-preset"]
}
```
Everything works when the same babel configuration is applied in the old `.babelrc` format.

I traced down the problem by inserting console.logs in `node_modules/@babel/register/lib/node.js`, function `compile`. When using `babel.config.js` you can see that all plugins of `module:metro-react-native-babel-preset` are being applied to the `node_modules/react-native/local-cli/core/index.js` file; when using `.babelrc` that's not the case. The code breaks because apparently `regeneratorRuntime` call is inserted, but `babel/polyfill` (or `regenerator-runtime/runtime`) import is not happening. 

[Repository with working state](https://github.com/jonathanglasmeyer/react-native-57-rc4-babel-config-bug-repo/commit/56baec58dab39d67507be43566259db8e0a19692)
[With a broken state](https://github.com/jonathanglasmeyer/react-native-57-rc4-babel-config-bug-repo/tree/ee4497c2e2856ad6c52d632c73ad0e038d5b7d03)

Is this behaviour known? How are users expected to make use of `babel.config.js`?

**Comment by kelset** [comment]
@jonathanglasmeyer open a dedicated issue for that.

**Comment by michaelknoch** [comment]
i tried to run my app with rn 0.57rc 4 but it seems that weird things are happening when transforming `local-cli/server/server.js`. I migrated from babel-preset-react-native to metro-react-native-babel-preset

```js
> node node_modules/react-native/local-cli/cli.js start

/Users/michaelknoch/dev/flow/smartphone-app/node_modules/@babel/core/lib/transformation/normalize-file.js:209
    throw err;
    ^

SyntaxError: /Users/michaelknoch/dev/flow/smartphone-app/node_modules/react-native/local-cli/server/server.js: Unexpected token, expected ";" (9:32)

   7 |         })
   8 |       );
>  9 | /* remotedev-server end */ code is licensed under the MIT license found in the
     |                                 ^
  10 |  * LICENSE file in the root directory of this source tree.
  11 |  *
  12 |  * @format

```

**Comment by Titozzz** [comment]
@michaelknoch FYI: This should be posted directly into react-native repo in the issues tab. You should remove your node_modules and yarn again. They seem to be corrupted.

**Comment by barrault01** [comment]
Hi there, I think it is really important to make https://github.com/facebook/react-native/commit/73a8aa2cbf29fc7da2fcb91a6177273bd1e3b610 cherry-picked into the next release. It's a bug that disallows RN to be integrated inside a swift project using cocoapods.

**Comment by ocarreterom** [comment]
Please add this commit: https://github.com/facebook/react-native/commit/9a4c21f92bf7a979d1719dd9dad533fb246786ca
The app build is broken in AppCenter 😞

**Comment by turnrye** [comment]
@ocarreterom @barrault01 looks like both of these are in 0.57 already

**Comment by hramos** [comment]
@barrault01 as stated multiples times on this thread, please limit cherry-pick requests to commits already in master. That commit is from a pull request that is not yet merged (https://github.com/facebook/react-native/pull/18050).

**Comment by hramos** [comment]
I just cherry-picked the following (kudos to @kelset for doing the work to ensure these could land cleanly):

1. https://github.com/facebook/react-native/commit/158e9c4ddeeff5a6e561cc9612036998f8b488cd
2. https://github.com/facebook/react-native/commit/d0eb8ff858f45f5d8315860ad91807ad0e685437
3. https://github.com/facebook/react-native/commit/f5be89b8b995bd6a6e040ccde83adb22ab4d273c
4. https://github.com/facebook/react-native/commit/672528ffde3b467ccdfd6b1ce0352f150b20c922
5. https://github.com/facebook/react-native/commit/5491c3f942430982ce9cb6140ed1733879ed3d1d
6. https://github.com/facebook/react-native/commit/2307ea60d03edd234511bfe32474c453f30c1693 (cc @rowinbot)
7. https://github.com/facebook/react-native/commit/0bf56e0746c1e60204996558b15ba425b12ed457
8. As a bonus, https://github.com/facebook/react-native/commit/eef8d47a37211bf7d4978db75df1fedd9cacbde8 requested by @barrault01 made it in as the PR they referred to finally landed today.

**Comment by TheSavior** [comment]
Apparently flow is failing due to a commit from a few weeks ago that got reverted but that revert commit didn’t end up on the release branch.

This commit should get picked: https://github.com/facebook/react-native/commit/d2c27f5bff750dbb3d04abf6b26a9d2f1ac7d0b5

And potentially the parent of that commit too if you have merge conflicts.

**Comment by hramos** [comment]
Thanks Eli. `0.57-stable` is now green, and ready to go: https://circleci.com/workflow-run/fb39f98d-de6a-4738-847d-65963a143cf9

@grabbou / @kelset: feel free to promote to stable when you're ready! The release should make it onto npm after running the bump version script, as the publish_to_npm step should run successfully without my involvement once it picks up the `v0.57.0` git tag in `0.57-stable`.

**Comment by grabbou** [comment]
Thanks for conducting this process @hramos. I am doing release right now.

**Comment by ngraef** [comment]
What's the process for releasing a new version of `react-native-git-upgrade`? See https://github.com/facebook/react-native/pull/20790 for context.

Regardless of whether the fix is published or not, I think this deserves a mention in the Upgrade section of the 0.57 changelog with information to update/avoid/patch `react-native-git-upgrade`. (Sorry I didn't notice the omission before release.)

**Comment by kelset** [comment]
> releasing a new version of react-native-git-upgrade

I think that it would be great to have someone to maintain it "properly", at the moment is not being actively maintained AFAIK. Last time we spoke about it was during a [core meeting](https://github.com/react-native-community/discussions-and-proposals/blob/master/core-meetings/2018-08-cli-meeting.md#upgrade-and-react-native-git-upgrade).

That said, I think that it'd be better to open a dedicated issue to discuss its future [here](https://github.com/react-native-community/discussions-and-proposals/issues) so that we can keep this issue on-topic.

**Comment by fungilation** [comment]
`react-native-git-upgrade` should get a version bump on npm right now before discussion on its maintainability, as it's broken right now for anyone using it to upgrade to 0.57.

**Comment by markharding** [comment]
Can the following patch be applied for 0.57.1 https://github.com/facebook/react-native/issues/20797?

**Comment by jamsch** [comment]
Just noticed that [ReactFragmentActivity](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/ReactFragmentActivity.java) says it will be removed on 0.57. Will this still be removed?

**Comment by empyrical** [comment]
I would love if this commit made it into 0.57.1:

https://github.com/facebook/react-native/commit/4b106be47765dd391f7a4cc6cf0e705ae977b90a

**Comment by hramos** [comment]
@jamsch it didn't end up getting removed. We should probably remove that comment, as we have not formalized how deprecations should be handled (as in, how many releases will we allow them to remain @deprecated before outright removing them).

**Comment by hramos** [comment]
@markharding we're only cherry-picking commits from `master`. That PR is still open.

**Comment by kelset** [comment]
As a rough estimate for 0.57.1, currently we are focusing on having XCode 10 working properly with RN (since they just released it) plus a Metro fix before releasing it.

Aside from that, please only ask for commits already merged to master - as probably we will only cut a first 0.58 rc after we release 0.57.1.

**Comment by rafeca** [comment]
The fix for Metro already landed in master (the change is only in the React Native repo). 
This is the commit: https://github.com/facebook/react-native/commit/7a69f1aa272a9b71755033a80f6f4aa5e9dcbaf6 and it would be lovely if it's cherry-picked to v0.57.1

**Comment by hramos** [comment]
Xcode fix: I nominate https://github.com/facebook/react-native/commit/8103c431c897c02d47cfad1e71bb2e6ddaabbdc0 for 0.57.1. That commit is in master, but did not make the cut for 0.57.0.

**Comment by hawkrives** [comment]
I'd like to nominate https://github.com/facebook/react-native/commit/b9c28c236bc971a5fbc51a3bda09c3980d547b96 for 0.57.1; without it, I have to `// $FlowExpectedError` whenever I use `<TextInput scrollEnabled={false} />`. (I think it's all ready to go; apologies if it's not.)

**Comment by oblador** [comment]
I’d like to nominate https://github.com/facebook/react-native/commit/9fca769e7666df696299b422c140d6509e726ec6 to fix storybook compatibility.

**Comment by grabbou** [comment]
Currently `react-native init` is broken - [this issue](https://github.com/facebook/react-native/issues/21219 ) should be fixed ASAP in next release.

**Comment by christianbach** [comment]
Perhaps #21232 is suited for the 0.57.1 release?

**Comment by kelset** [comment]
Hey everyone, the current idea is to do a small 0.57.1 before the weekend that won't fix all the Xcode 10 issues but at least give us all those nominated commits plus a couple more fixes - we just run in a Metro issue yesterday evening so we are waiting for a new release of that before proceeding.

**Comment by kelset** [comment]
Update: since Metro `0.45.6` has been released and works great, I've tested locally this set of cherry picks (in order of picking):


- [Expose scrollEnabled as iOS prop for TextInput](https://github.com/facebook/react-native/commit/b9c28c236bc971a5fbc51a3bda09c3980d547b96)
- [fix RCTNetInfo first time connection status](https://github.com/facebook/react-native/commit/36199d3dda9eedb45628fc49483e98764d67847a)
- [Give RNPM the ability to look for plugins in `@Scoped` modules](https://github.com/facebook/react-native/commit/4b106be47765dd391f7a4cc6cf0e705ae977b90a)
- [Making RCTIsIPhoneX() return true for the R and Max models](https://github.com/facebook/react-native/commit/5e7c3ca0057f6084d692e30ae4db863fb20968ff)
- [Upgrade babel-eslint to 9.0.0](https://github.com/facebook/react-native/commit/44dc283bcd0a75826d9be86cdc727e32d5697ef2)
- [Pass the maxWorkers config param correctly to Metro](https://github.com/facebook/react-native/commit/7a69f1aa272a9b71755033a80f6f4aa5e9dcbaf6)
- [Fix ignored --projectRoot/watchFolders arguments](https://github.com/facebook/react-native/commit/9fca769e7666df696299b422c140d6509e726ec6)
- [Expose AllowFileAccess property in WebView](https://github.com/facebook/react-native/commit/0c576ef84a1c7f79b228f205cc687ab1b945bda1)
- [Way to register RCT_MODULE in Plugin2.0 instead of +load](https://github.com/facebook/react-native/commit/5c160e5dedae713c686d88d4b9d4308b596e68a7)
- [Debug only code were leaking into release builds on iOS](https://github.com/facebook/react-native/commit/d1ff0b0cc51c31cae89689b2ad2f4b35f29531d8)
- [Update RCTLinkingManager.h to explicitly state the 'nullability' of parameters](https://github.com/facebook/react-native/commit/2271d1f912435eba7da2414ea4665ba8e56c7ad7)

Plus, on top, the commit to "enforce" the newer metro [Bump metro@0.45.6](https://github.com/facebook/react-native/commit/023f0a68555f7629fa51aa38b0aabea60c7b6c36) (which caused a conflict, I think because I haven't cherry picked the other 3/4 commits that bump the minor version - I just solved it locally).

With this "local set" of commits on top of `0.57-stable` all my tests were ✅

I'm just waiting for the approval from other maintainers then we can proceed to release `0.57.1`.

---

Minor notes on a couple more commits:
- for some reason didn't manage to cherry pick [Fix TextInput.setNativeProps({text: ''}) to work](https://github.com/facebook/react-native/commit/2307ea60d03edd234511bfe32474c453f30c1693) -  not sure why
- the Xcode related commit [Removing development team from Xcode project.](https://github.com/facebook/react-native/commit/8103c431c897c02d47cfad1e71bb2e6ddaabbdc0) caused conflicts that I couldn't fix even by importing earlier commits like [Dynamically load WebKit](https://github.com/facebook/react-native/commit/28b058c341690bd35e1d59885762ec29614a3d45) and [Implement 'dataDetectorTypes' prop](https://github.com/facebook/react-native/commit/1af17f164828b6d6fa0450af46baf945745363e7), so I gave up. But should not too big of a deal since it wasn't a "all mighty fix" for Xcode 10 anyway.

**Comment by chrisnojima** [comment]
Would it be possible to re-attempt to get https://github.com/facebook/react-native/commit/2307ea60d03edd234511bfe32474c453f30c1693 in? it's a tiny diff in one file

**Comment by rowinbot** [comment]
Isn't [facebook/react-native@2307ea6](https://github.com/facebook/react-native/commit/2307ea60d03edd234511bfe32474c453f30c1693) packed in `0.57.0` I've been using it for a week now and it seems to work already

**Comment by hramos** [comment]
Yes, https://github.com/facebook/react-native/commit/2307ea60d03edd234511bfe32474c453f30c1693 is already in 0.57, even though the tags on that commit do not show it: https://github.com/facebook/react-native/commits/0.57-stable

**Comment by mjmasn** [comment]
If there's going to be a 0.57.2 it would be awesome to get these two fixes for scroll views in:
facebook/react-native@e0170a9
facebook/react-native@4af4da9

**Comment by DaveCole** [comment]
Hey there - I followed the instructions on the changelog to update everything to 0.57.1 and am now getting the "No bundle URL present." even though `react-native run-ios` succeeds in building.  I'm guessing it has something to do with the babel change but I'm a little lost on what to do next.  Any help would be great, thanks.

**Comment by jhaenchen** [comment]
I was told by a peer that this version of react-native breaks symlinks. Is that accurate? My symlinks are definitely not working...

**Comment by kelset** [comment]
@jhaenchen open a dedicated issue in the main repo.

**Comment by sunnylqm** [comment]
Please include this fix in 0.57.2 https://github.com/facebook/react-native/commit/1323acd0bb23eb5df76ee6561ba86f4802b7432f

**Comment by kelset** [comment]
Hey folks, since with the core we agreed that 0.57.x is a fairly stable release and that we are waiting on a few things from FB internal codebase to land to the OSS for 0.58, we decided to "extend" the lifespan of 0.57.

For this reason, I've opened a new issue for version 0.57.1 (latest currently) to which suggest cherry picks. And similarly when 0.57.2 will be released we'll do the same.

**Comment by buixuanthe2212** [comment]
borderTopColor,  borderLeftColor, borderRightColor, borderBottomColor
not working for react-native 0.57
;(
